python/lsst/meas/astrom/astrometry.py

```diff
@@ -417,20 +417,24 @@ def _removeMagnitudeOutliers(self, sourceFluxField, refFluxField, matchesIn):
             refMag[i] = (match[0][refFluxField]*units.nJy).to_value(units.ABmag)
 
         deltaMag = refMag - sourceMag
-        zp = np.median(deltaMag)
+        # Protect against negative fluxes and nans in the reference catalog.
+        goodDelta, = np.where(np.isfinite(deltaMag))
+        zp = np.median(deltaMag[goodDelta])
         # Use median absolute deviation (MAD) for zpSigma
         # Also require a minimum scatter to prevent floating-point errors from
         # rejecting objects in zero-noise tests.
-        zpSigma = np.clip(1.4826*np.median(np.abs(deltaMag - zp)), 1e-3, None)
+        zpSigma = np.clip(1.4826*np.median(np.abs(deltaMag[goodDelta] - zp)), 1e-3, None)
```

Comment:
Can we replace this whole calculation with `scipy.stats.median_abs_deviation`, which has a `nan_policy` option?

Reply:
See above about protecting against nans being insufficient.  But I can use the scipy stats version instead of rolling my own.